### PR TITLE
Overhauled DiscordGuild:

### PR DESCRIPTION
**Pull request by Plerx2493**
- Created methods for operations only relying on ids
- Fixed some xml docs with old enum names
- Fixed some nullability annotations

### CODE REVIEW

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 1]
```diff
@@ -923,29 +968,29 @@ public async Task RemoveMemberAsync(ulong userId, string? reason = null)
...
     /// <param name="reason">Reason for audit logs.</param>
     /// <returns></returns>
     /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="DiscordPermissions.BanMembers"/> permission.</exception>
     /// <exception cref="NotFoundException">Thrown when the member does not exist.</exception>
     /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
     /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-    public async Task BanMemberAsync(ulong user_id, int delete_message_days = 0, string reason = null)
-        => await Discord.ApiClient.CreateGuildBanAsync(Id, user_id, delete_message_days, reason);
+    public async Task BanMemberAsync(ulong userId, int deleteMessageDays = 0, string? reason = null)
+        => await Discord.ApiClient.CreateGuildBanAsync(Id, userId, deleteMessageDays, reason);
```
Comment (akiraveliara) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:993]: should be changed to deleteMessageSeconds, and we should put up a breaking change warning

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 2]
```diff
@@ -802,10 +829,28 @@ public async Task<IReadOnlyList<DiscordMember>> SearchMembersAsync(string name,
...
+    /// <param name="accessToken">User's access token (OAuth2)</param>
+    /// <param name="nickname">new nickname</param>
+    /// <param name="roles">new roles</param>
+    /// <param name="muted">whether this user has to be muted</param>
+    /// <param name="deaf">whether this user has to be deafened</param>
+    /// <returns>Only returns the member if they were not already in the guild</returns>
+    /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="DiscordPermissions.CreateInstantInvite" /> permission.</exception>
+    /// <exception cref="NotFoundException">Thrown when the <paramref name="userId"/> or <paramref name="accessToken"/> is not found.</exception>
+    /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
+    /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
```
Comment (akiraveliara) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:849]: doc comments should be full sentences, ending in a period

**Review by akiraveliara — CHANGES_REQUESTED** [review]
the doc comment thing goes for most of the diff, but it's not much of a blocker imo

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 3]
```diff
@@ -740,7 +751,23 @@ public async Task<IReadOnlyList<DiscordScheduledGuildEvent>> GetEventsAsync(bool
...
     /// <param name="before">Fetch users before this id. Mutually exclusive with after</param>
-    public async IAsyncEnumerable<DiscordUser> GetEventUsersAsync(DiscordScheduledGuildEvent guildEvent, int limit = 100, ulong? after = null, ulong? before = null)
+    public IAsyncEnumerable<DiscordUser> GetEventUsersAsync
+    (
+        DiscordScheduledGuildEvent guildEvent,
+        int limit = 100,
+        ulong? after = null,
+        ulong? before = null
+    )
+        => GetEventUsersAsync(guildEvent.Id, limit, after, before);
```
Comment (OoLunar) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:761]: It feels like something here should be inlined

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 4]
```diff
@@ -802,10 +829,28 @@ public async Task<IReadOnlyList<DiscordMember>> SearchMembersAsync(string name,
     /// <exception cref="NotFoundException">Thrown when the <paramref name="user"/> or <paramref name="accessToken"/> is not found.</exception>
     /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
     /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-    public async Task<DiscordMember?> AddMemberAsync(DiscordUser user, string accessToken, string nickname = null, IEnumerable<DiscordRole> roles = null,
+    public async Task<DiscordMember?> AddMemberAsync(DiscordUser user, string accessToken, string? nickname = null, IEnumerable<DiscordRole>? roles = null,
         bool muted = false, bool deaf = false)
         => await Discord.ApiClient.AddGuildMemberAsync(Id, user.Id, accessToken, muted, deaf, nickname, roles);
```
Comment (OoLunar) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:834]: This doesn't feel right either

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 5]
```diff
@@ -802,10 +829,28 @@ public async Task<IReadOnlyList<DiscordMember>> SearchMembersAsync(string name,
...
+    /// <param name="muted">whether this user has to be muted</param>
+    /// <param name="deaf">whether this user has to be deafened</param>
+    /// <returns>Only returns the member if they were not already in the guild</returns>
+    /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="DiscordPermissions.CreateInstantInvite" /> permission.</exception>
+    /// <exception cref="NotFoundException">Thrown when the <paramref name="userId"/> or <paramref name="accessToken"/> is not found.</exception>
+    /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
+    /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+    public async Task<DiscordMember?> AddMemberAsync(ulong userId, string accessToken, string? nickname = null, IEnumerable<DiscordRole>? roles = null,
+    bool muted = false, bool deaf = false)
+        => await Discord.ApiClient.AddGuildMemberAsync(Id, userId, accessToken, muted, deaf, nickname, roles);
```
Comment (OoLunar) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:852]: +1

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 6]
```diff
@@ -967,7 +1012,7 @@ public async Task<DiscordBulkBan> BulkBanMembersAsync(IEnumerable<DiscordUser> u
     /// <param name="deleteMessageSeconds">Timespan in seconds to delete messages from the banned users</param>
     /// <param name="reason">Reason for audit logs.</param>
     /// <returns>Response contains a which users were banned and which were not.</returns>
-    public async Task<DiscordBulkBan> BulkBanMembersAsync(IEnumerable<ulong> userIds, int deleteMessageSeconds = 0, string reason = null)
+    public async Task<DiscordBulkBan> BulkBanMembersAsync(IEnumerable<ulong> userIds, int deleteMessageSeconds = 0, string? reason = null)
     => await Discord.ApiClient.CreateGuildBulkBanAsync(Id, userIds, deleteMessageSeconds, reason);
```
Comment (OoLunar) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:1016]: Indent

File: DSharpPlus/Entities/Guild/DiscordGuild.cs (hunk 2 above)
Comment (Plerx2493) [reply DSharpPlus/Entities/Guild/DiscordGuild.cs:849]: Currently not in the mood to rewrite all xmldocs in that file :when:

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 7]
```diff
@@ -1259,6 +1317,22 @@ public async Task<int> GetPruneCountAsync(int days = 7, IEnumerable<DiscordRole>
...
+    /// <param name="reason">Reason for audit logs.</param>
+    /// <returns>Number of users pruned.</returns>
+    /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="DiscordPermissions.ManageChannels"/> permission.</exception>
+    /// <exception cref="NotFoundException">Thrown when the guild does not exist.</exception>
+    /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
+    /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+    public async Task<int?> PruneAsync(int days = 7, bool computePruneCount = true, IEnumerable<ulong>? includedRoleIds = null, string? reason = null)
+        => await Discord.ApiClient.BeginGuildPruneAsync(Id, days, computePruneCount, includedRoleIds?.Where(x => _roles.ContainsKey(x)), reason);
+
+
```
Comment (OoLunar) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:1335]: Double newline

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 8]
```diff
@@ -1737,6 +1853,19 @@ public async Task DeleteEmojiAsync(DiscordGuildEmoji emoji, string? reason = nul
...
+    /// <param name="emojiId">Emoji to delete.</param>
+    /// <param name="reason">Reason for audit log.</param>
+    /// <returns></returns>
+    /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="DiscordPermissions.ManageEmojis"/> permission.</exception>
+    /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+    /// <exception cref="NotFoundException">Thrown when the emoji does not exist on this guild</exception>
+    public async Task DeleteEmojiAsync(ulong emojiId, string? reason = null)
+        => await Discord.ApiClient.DeleteGuildEmojiAsync(Id, emojiId, reason);
+
+
```
Comment (OoLunar) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:1868]: Double newline

File: DSharpPlus/Entities/Guild/DiscordGuild.cs (hunk 1 above)
Comment (OoLunar) [reply DSharpPlus/Entities/Guild/DiscordGuild.cs:993]: Should we just change it to a timespan and pass `TotalSeconds` to the API?

Comment (Plerx2493) [reply DSharpPlus/Entities/Guild/DiscordGuild.cs:993]: Yeah, im on it

Comment (akiraveliara) [reply DSharpPlus/Entities/Guild/DiscordGuild.cs:993]: considered doing that for v6, we can totally do that already

File: DSharpPlus/Entities/Guild/DiscordGuild.cs (hunk 3 above)
Comment (Plerx2493) [reply DSharpPlus/Entities/Guild/DiscordGuild.cs:761]: The `=>` part or everything? The line is pretty long

**Review by OoLunar — APPROVED** [review]

**Review by akiraveliara — APPROVED** [review]

**Comment by akiraveliara** [comment]
list this as a break in the tracking issue and announcements

**Pull request merged**